### PR TITLE
tsconfig builds

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #7407 

## Description of Changes
- Added a new `tsconfig.build.json` file for every package/module/lib
- Refactored all references to point to the new tsconfigs
- Modified the base tsconfig to include folders that aren't included in the final build such as "/test" and "__mock__"

## Test Plan
- Run various build and type-check commands you usually use in your workflow

## Deployment Plan

1. 

## Other Considerations

### CODE REVIEW

File: libs/adapters/tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
   "compilerOptions": {
-    "rootDir": "src",
-    "outDir": "build",
+    "rootDir": ".",
     "tsBuildInfoFile": "build/.tsbuildinfo",
     "strict": true
   },
   "include": ["src"],
-  "references": [{ "path": "../core" }, { "path": "../logging" }]
+  "references": [
```
Comment (Rotorsoft) [inline libs/adapters/tsconfig.json:9]: did you try without references here? not sure this is needed when not building

File: libs/adapters/tsconfig.json [hunk 2]
```diff
@@ -1,11 +1,13 @@
 {
   "extends": "../tsconfig.json",
   "compilerOptions": {
-    "rootDir": "src",
-    "outDir": "build",
+    "rootDir": ".",
     "tsBuildInfoFile": "build/.tsbuildinfo",
```
Comment (Rotorsoft) [inline libs/adapters/tsconfig.json:5]: tsBuildInfoFile is also a build thing right?

File: libs/adapters/tsconfig.build.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "src",
+    "outDir": "build"
```
Comment (Rotorsoft) [inline libs/adapters/tsconfig.build.json:5]: no strict when building?

File: libs/adapters/tsconfig.json (hunk 2 above)
Comment (timolegros) [reply libs/adapters/tsconfig.json:5]: Removing this from the base tsconfig is only possible if we also remove the references. As described in the issue above, this isn't possible without some concessions.

File: libs/adapters/tsconfig.build.json (hunk 3 above)
Comment (timolegros) [reply libs/adapters/tsconfig.build.json:5]: I believe the `strict` property will be inherited from the base tsconfig.

File: libs/adapters/tsconfig.json (hunk 1 above)
Comment (timolegros) [reply libs/adapters/tsconfig.json:9]: TBH we could move `adapters/src/index.ts` to `/adapters/index.ts` and we wouldn't need the references. Not sure how you feel about moving that file tho.

**Review by Rotorsoft — APPROVED** [review]

**Pull request merged**